src/redux/chromeReducers.ts

```diff
@@ -338,7 +345,56 @@ export function toggleNotificationsReducer(state: ChromeState) {
     ...state,
     notifications: {
       ...state.notifications,
+      data: testData, // This is where we would pass the data instead of simply hardcoding it in
       isExpanded: !state.notifications?.isExpanded,
     },
   };
 }
+
+export function notificationsReducer(state: ChromeState, action: { type: string; payload?: number }): ChromeState {
+  console.log('Looking at our state bfore updating: ', state.notifications?.data);
+  switch (action.type) {
+    case MARK_NOTIFICATION_AS_READ:
+      return {
+        ...state,
+        notifications: {
+          isExpanded: state.notifications?.isExpanded || false,
+          count: state.notifications?.count || 0,
+          data: (state.notifications?.data || []).map((notification: NotificationData) =>
+            notification.id === action.payload ? { ...notification, read: true } : notification
+          ),
+        },
+      };
+    case MARK_NOTIFICATION_AS_UNREAD:
+      return {
+        ...state,
+        notifications: {
+          isExpanded: state.notifications?.isExpanded || false,
+          count: state.notifications?.count || 0,
```

Comment:
No need to re-calculate these values. It should be enough to spread the `notifications` from state
```suggestion
          ...state.notifications
```